### PR TITLE
 Save custom value in SelectBox after focus disappearance (T708199)

### PR DESCRIPTION
**Pull request by KsyushaM**

### CODE REVIEW

File: js/ui/select_box.js [hunk 1]
```diff
@@ -591,14 +591,26 @@ var SelectBox = DropDownList.inherit({
         this._setPopupOption("width");
     },
 
+    _sameSelectedItem: function() {
```
Comment (churkin) [inline js/ui/select_box.js]: I dont understand what this name means

File: js/ui/select_box.js [hunk 2]
```diff
@@ -591,14 +591,26 @@ var SelectBox = DropDownList.inherit({
...
+        var initialSelectedItem = this.option("selectedItem"),
```
Comment (churkin) [inline js/ui/select_box.js]:
```javascript
var initialSelectedItem = this.option("selectedItem");
var value = this._displayGetter(initialSelectedItem);
var displayValue = value ? String(value) : '';
var searchValue = this._searchValue();

return displayValue === searchValue;
```

**Review by churkin — CHANGES_REQUESTED** [review]

File: js/ui/select_box.js (hunk 1 above)
Comment (KsyushaM) [reply js/ui/select_box.js]: OK

File: js/ui/select_box.js (hunk 2 above)
Comment (KsyushaM) [reply js/ui/select_box.js]: OK

**Review by mrjohn0011 — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**